client/asset/eth/nodeclient_harness_test.go

```diff
@@ -251,19 +393,41 @@ func setupWallet(walletDir, seed, listenAddress string) error {
 		Pass:     []byte(pw),
 		Settings: settings,
 		DataDir:  walletDir,
-		Net:      dex.Simnet,
+		Net:      net,
 	}
 	return CreateWallet(&createWalletParams)
 }
 
 func syncClient(cl *nodeClient) error {
+	giveUpAt := 60
+	if testnet {
+		giveUpAt = 10000
+	}
 	for i := 0; ; i++ {
+		if err := ctx.Err(); err != nil {
+			return err
+		}
 		prog := cl.syncProgress()
-		if prog.CurrentBlock >= prog.HighestBlock {
-			return nil
+		if testnet {
+			syncing := prog.CurrentBlock > prog.HighestBlock || prog.CurrentBlock == 0
```

Comment:
I'm confused by `prog.CurrentBlock > prog.HighestBlock` here.  For simnet we declare synced (not syncing) as `prog.CurrentBlock >= prog.HighestBlock && prog.CurrentBlock != 0`.
Should this here be a `<` to indicate styncing?

Reply:
I'm also a little confused here. I beleive `HighestBlock` is our local synced block, and `CurrentBlock` is the best block from the perspective of a peer. CurrentBlock always starts at zero I believe, when we have no peers. This is correct for syncing I think, after a bit of observing sure seems that way. If current block has reached highest block, we are synced and no longer syncing. I keep looking at SyncStatus thinking somethings wrong but I don't want to touch it in this pr.

Looking again, the simnet version should just be `==` I think as we dont want to worry about the header being old there.

Reply:
> I beleive HighestBlock is our local synced block, and CurrentBlock is the best block from the perspective of a peer

Pretty sure it's the opposite.  https://ethereum.stackexchange.com/a/424/90199

Reply:
With `fmt.Printf("client: %d\ncurrent: %d\nhighest: %d\nstarting: %d\n", clientNum, prog.CurrentBlock, prog.HighestBlock, prog.StartingBlock)`

See:
```
Testnet nodes starting sync, this may take a while...
client: 2
current: 6619110
highest: 0
starting: 0
client: 1
current: 6619110
highest: 0
starting: 0
client: 2
...
client: 1
current: 6619110
highest: 6666234
starting: 6619110
INFO [04-06|14:21:27.332] Imported new block headers               count=192 elapsed=115.877ms number=6,619,302 hash=de8596..cfb8c4 age=1w1d3h
INFO [04-06|14:21:27.533] Imported new block headers               count=192 elapsed=32.827ms  number=6,619,494 hash=a119dc..f08c74 age=1w1d2h
INFO [04-06|14:21:27.818] Imported new block headers               count=192 elapsed=33.990ms  number=6,619,686 hash=9c59c7..50f588 age=1w1d2h
client: 2
current: 6619110
highest: 0
starting: 0
...
client: 1
current: 6666242
highest: 6666242
starting: 6619110
...
client: 2
current: 6619110
highest: 6666272
starting: 6619110
...
client: 2
current: 6666279
highest: 6666279
starting: 6619110
```

Reply:
So, I think you are right. My way was just working because prog.CurrentBlock == prog.HighestBlock would still be not syncing... Otherwise the header check was stopping it?